### PR TITLE
add abbreviation problem in dynamic

### PR DESCRIPTION
**Pull request by duongoku**
# Abbreviation Problem
Source: [Hackerrank](https://www.hackerrank.com/challenges/abbr/problem)

## Table of Contents
- [Abbreviation Problem](#abbreviation-problem)
  - [Table of Contents](#table-of-contents)
  - [Source Code](#source-code)
  - [Graphs](#graphs)
    - [Flowchart](#flowchart)
    - [Control Flow Graph](#control-flow-graph)
    - [Data Flow Graph](#data-flow-graph)
  - [All-DU-Paths coverage testcases](#all-du-paths-coverage-testcases)
  - [MCC coverage testcases](#mcc-coverage-testcases)
  - [Other testcases](#other-testcases)

## Source Code
carbon

### Flowchart
flowchart
### Control Flow Graph
controlflow
### Data Flow Graph
dataflow

## All-DU-Paths coverage testcases
| Path | a | b | Expected | Actual | Verdict |
|---|---|---|---|---|---|
| 1-> 2-6 -> 8 -> 21 | "" | "" | true | true | Pass |
| 1-> 2-6 -> 8 -> 9 -> 10 -> 11 -> 14 -> 15 ... | "a" | "" | true | true | Pass |
| 1-> 2-6 -> 8 -> 9 -> 10 -> 11 -> 14 -> 15 ... | "a" | "b" | false | false | Pass |
| 1-> 2-6 -> 8 -> 9 -> 10 -> 11 -> 12 -> 14 ... | "a" | "a" | false | false | Pass |
| 1-> 2-6 -> 8 -> 9 -> 10 -> 11 -> 12 -> 14 ... | "A" | "A" | true | true | Pass |

## MCC coverage testcases
Line column show the truth value of the condition of corresponding line: F stands for False and T stands for true.

| Path | Line 8 | Line 9 | Line 10 | Line 11 (j < len(b)) | Line 11 (other one) | Line 14 |  a | b | Expected | Actual | Verdict |
|---|---|---|---|---|---|---|---|---|---|---|---|
| 1 -> 2-7 -> 8 -> 20-22 | F | unknown | unknown | unknown | unknown | unknown | "" | "" | true | true | Pass |
| 1 -> 2-7 -> 8 -> 9 -> 10 -> 11 -> 14 ... | T | T | T | F | unknown | unknown | "a" | "" | true | true | Pass |
| 1 -> 2-7 -> 8 -> 9 -> 10 -> 11 -> 14 ... | F | T | T | T | F | unknown | "a" | "a" | false | false | Pass |
| 1 -> 2-7 -> 8 -> 9 -> 10 -> 11 -> 14 ... | F | T | T | T | F | unknown | "a" | "b" | false | false | Pass |
| 1 -> 2-7 -> 8 -> 9 -> 10 -> 11 -> 12 ... | F | T | T | T | T | T | "A" | "A" | true | true | Pass |

## Other testcases
| a | b | Expected | Actual | Verdict |
|---|---|---|---|---|
| "uOHlGMdUBc" | "uOalGMdUBCasdcavsdf" | false | false | pass |
| "kotgDIUagj" | "DIU" | true | true | pass |
| "WPTffVkSNl" | "WPTVSN" | true | true | pass |
| "CoJsPURrVX" | "CPUVX" | false | false | pass |
| "xasreDHndqvCnFfX" | "DHndqvCnFX" | false | false | pass |
| "XFEaWCxpeepGjOnCCsFh" | "XFEAWCPEPGOCCSF" | true | true | pass |

### CODE REVIEW

File: dynamic/abbreviation_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package dynamic
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestAbbreviation(t *testing.T) {
+	aStrings := []string{
```
Comment (raklaptudirm) [inline dynamic/abbreviation_test.go:9]: Convert the three different slices to a single struct slice:
```go
tests := []struct{
  a, b string
  expected bool
}{}
```

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: dynamic/abbreviation_test.go (hunk 1 above)
Comment (duongoku) [reply dynamic/abbreviation_test.go]: Done!

**Comment by duongoku** [comment]
seems like the CI has some problem with linting 🤔

**Comment by raklaptudirm** [comment]
Run `gofmt`

**Comment by duongoku** [comment]
"level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: goanalysis_metalinter: buildir: package \"sort_test\" (isInitialPkg: true, needAnalyzeSource: true): in github.com/TheAlgorithms/Go/sort_test.TestBubble: cannot convert github.com/TheAlgorithms/Go/sort.Bubble (func[T github.com/TheAlgorithms/Go/constraints.Ordered](arr []T) []T) to func([]int) []int\n\n" github CI said this, it has problem with the bubblesort 🤔

**Comment by raklaptudirm** [comment]
Ah, then thats ok.

**Comment by raklaptudirm** [comment]
@duongoku Looks like the others are busy. I will give them a heads up.
/cc @siriak @tjgurwara99

**Comment by duongoku** [comment]
Thank you very much

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**